airflow/contrib/example_dags/example_gcf_function_deploy_invoke_delete.py

```diff
@@ -0,0 +1,109 @@
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+Example Airflow DAG that displays interactions with Google Cloud Functions.
+It creates a function, invokes and then deletes it.
+
+This DAG relies on the following Airflow variables
+https://airflow.apache.org/concepts.html#variables
+* PROJECT_ID - Google Cloud Project to use for the Cloud Function.
+* REGION - Google Cloud Functions region where the function should be
+  created.
+* ENTRYPOINT - Name of the executable function in the source code.
+* and one of the below:
+    - SOURCE_ARCHIVE_URL - Path to the zipped source in Google Cloud Storage
+    or
+    (
+        - SOURCE_UPLOAD_URL - Generated upload URL for the zipped source
+        and
+        - ZIP_PATH - Local path to the zipped source archive
+    )
+    or
+    - SOURCE_REPOSITORY - The URL pointing to the hosted repository where the function is
+    defined in a supported Cloud Source Repository URL format
+    https://cloud.google.com/functions/docs/reference/rest/v1/projects.locations.functions#SourceRepository
+"""
+
+import datetime
+
+import airflow
+from airflow import models
+from airflow.contrib.operators.gcf_function_delete_operator import GCFFunctionDeleteOperator  # nopep8
+from airflow.contrib.operators.gcf_function_deploy_operator import GCFFunctionDeployOperator  # nopep8
+from airflow.contrib.operators.gcf_function_invoke_operator import GCFFunctionInvokeOperator  # nopep8
+
+PROJECT_ID = models.Variable.get('PROJECT_ID', '')
+REGION = models.Variable.get('REGION', '')
+SOURCE_ARCHIVE_URL = models.Variable.get('SOURCE_ARCHIVE_URL', '')
+SOURCE_UPLOAD_URL = models.Variable.get('SOURCE_UPLOAD_URL', '')
+SOURCE_REPOSITORY = models.Variable.get('SOURCE_REPOSITORY', '')
+ZIP_PATH = models.Variable.get('ZIP_PATH', '')
+ENTRYPOINT = models.Variable.get('ENTRYPOINT', '')
+FUNCTION_NAME = 'projects/{}/locations/{}/functions/{}'.format(PROJECT_ID, REGION,
+                                                               ENTRYPOINT)
+RUNTIME = 'nodejs6'
+body = {
+    "name": FUNCTION_NAME,
+    "entryPoint": ENTRYPOINT,
+    "runtime": RUNTIME,
+    "httpsTrigger": {}
+}
+
+# [START howto_operator_gcf_invoke_args]
+default_args = {
+    'start_date': airflow.utils.dates.days_ago(1),
+    'project_id': PROJECT_ID,
+    'region': REGION
+}
+# [END howto_operator_gcf_invoke_args]
+
+if SOURCE_ARCHIVE_URL:
+    body['sourceArchiveUrl'] = SOURCE_ARCHIVE_URL
+elif SOURCE_REPOSITORY:
+    body['sourceRepository'] = {
+        'url': SOURCE_REPOSITORY
+    }
+elif ZIP_PATH:
+    body['sourceUploadUrl'] = SOURCE_UPLOAD_URL
+    default_args['zip_path'] = ZIP_PATH
+
+with models.DAG(
+    'example_gcf_function_deploy_invoke_delete',
+    default_args=default_args,
+    schedule_interval=datetime.timedelta(days=1)
+) as dag:
+    deploy_task = GCFFunctionDeployOperator(
+        body=body,
+        task_id="gcf_deploy_task"
+    )
+
+    # [START howto_operator_gcf_invoke]
```

Comment:
Those comments should be better placed (see the deploy_delete.dag). Then you can easily check if the documentation looks like by following instructions from CONTRIBUTING.md. I think it's better to include the whole with.models.DAG (and make a comment that it also contains deploy/delete - this makes the example much more useful.

Reply:
I'm not sure, I think it would be better to link only small parts of the code to each operator's documentation. Only the parts relevant to the operator in question. This way we limit the "noise" - if somebody is looking for how to use `invoke`, they probably shouldn't have to filter out the `delete` and `deploy` in the example.

On the other hand I agree that showing a wide context could be useful for easier understanding of how all of these operators work, but maybe we could simply add another section in the docs, like: "Using all GCF Operators"?

This way we have both brevity in descriptions of individual operators and the wider context.

Reply:
OK. I am fine with both approaches. Indeed it's more focused even if it does not 'compile' nor show the body contents, and has less of a distraction. I will even update the ones in deploy/delete ones to follow the same pattern. I recalled what the original problem was - the main problem which I had with the more "narrow" approach was that it did not show the "body" content which IMHO should be crucial part of the example. Without it, the example is rather useless. But I think we can find a way to show also body in the docs. I will discuss with you offline and maybe we will implement it now in both pull requests.